File: app/src/main/java/com/zeapo/pwdstore/mainscreen/recyclerview/FolderRecyclerAdapter.java
package com.zeapo.pwdstore.mainscreen.recyclerview;

import android.support.annotation.NonNull;
import android.view.View;

import com.zeapo.pwdstore.mainscreen.SelectFolderActivity;
import com.zeapo.pwdstore.mainscreen.fragments.SelectFolderFragment;
import com.zeapo.pwdstore.utils.PasswordItem;

import java.util.ArrayList;

public class FolderRecyclerAdapter extends EntryRecyclerAdapter {
    private final SelectFolderFragment.OnFragmentInteractionListener listener;

    // Provide a suitable constructor (depends on the kind of dataset)
    public FolderRecyclerAdapter(SelectFolderActivity activity, SelectFolderFragment.OnFragmentInteractionListener listener, ArrayList<PasswordItem> values) {
        super(activity, values);
        this.listener = listener;
    }

    @NonNull
    protected View.OnClickListener getOnClickListener(final ViewHolder holder, final PasswordItem pass) {
        return new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                listener.onFragmentInteraction(pass);
                notifyItemChanged(holder.getAdapterPosition());
            }
        };
    }

}

File: app/src/main/java/com/zeapo/pwdstore/ui/adapters/FolderRecyclerAdapter.kt
/*
 * Copyright © 2014-2020 The Android Password Store Authors. All Rights Reserved.
 * SPDX-License-Identifier: GPL-3.0-only
 */
package com.zeapo.pwdstore.ui.adapters

import com.zeapo.pwdstore.SelectFolderFragment
import com.zeapo.pwdstore.utils.PasswordItem

class FolderRecyclerAdapter(
    private val listener: SelectFolderFragment.OnFragmentInteractionListener
) : EntryRecyclerAdapter() {

    override fun onItemClicked(holder: PasswordItemViewHolder, item: PasswordItem) {
        listener.onFragmentInteraction(item)
        notifyItemChanged(holder.absoluteAdapterPosition)
    }
}
